astropy/modeling/functional_models.py

```diff
@@ -514,17 +510,6 @@ class Scale(Fittable1DModel):
     linear = True
     fittable = True
 
-    _input_units_strict = True
-
-    _input_units_allow_dimensionless = True
-
-    @property
-    def input_units(self):
-        if self.factor.unit is None:
-            return None
-        else:
-            return {'x': self.factor.unit}
-
```

Comment:
With this code removed the Scale model does not always work as expected. For example

```
scl = Scale(1*u.m)
scl(2*u.kg)
<Quantity 2.0 kg m>
```
I think it will work correctly if you keep this code in and in the evaluate function do:
```
# x is converted to the same units as factor by
# the time it reaches here.
if isinstance(x, u.Quantity):
     x = x.value
return factor * x
```

Reply:
This will also resolve [this concern in gwcs](https://github.com/spacetelescope/gwcs/pull/133#issuecomment-378745395).

Reply:
I am using the ability for scale to change units to use it to do a pixel scale operation.
```python
In [3]: s = Scale(10*u.arcsec/u.pix)

In [4]: s(10*u.pix)
Out[4]: <Quantity 100.0 arcsec>
```

To my mind if you do:
```python
scl = Scale(1*u.m)
scl(2*u.kg)
```
then I think you should get `<Quantity 2.0 kg m>`! 

If you want a dimensionless scale then you can do it explicitly with:

```python
In [5]: s = Scale(10*u.one)

In [6]: s(10*u.kg)
Out[6]: <Quantity 100.0 kg>
```

Reply:
Oh yes, we've discussed this before. The reason I come back to it is that it is counterintuitive to me
because I think a `Scale` operation should not change dimensions or units. Sure, it's nice that it works conveniently in this one use case but I am a little concerned that if we agree to do it this way it may be a problem in the future with other operations. A plate scale is not really a scaling operation in mathematical sense. That's why we have the equivalencies. And your example can be done by

```
scl = Scale (10*u.arcsec)
scl(10*u.pix, equivalencies={'x': u.pixel_scale(1*u.arcsec/u.pix)})
Out[14]: <Quantity 100.0 arcsec>
```

Perhaps it's best to ask on a mailing list what the behaviour should be in this case.

Regardless of the decision `input_units` should stay so that the gwcs issue with units is resolved.

Reply:
On the `input_units` thing w/r/t gwcs. I feel like (ab)using `input_units` for "this model supports units" is wrong, if we really need that, we should do it explicitly like we do for fitting with units.

I guess that my opinion on this is that it's just cleaner and more predictable to me to delegate the unit handing of `Scale` to `Quantity` it's not like we are excluding either use case, as you can always make the factor dimensionless.

Reply:
To me whether a use case can work or not is a technical issue. What we need to decide is whether it is correct for a Scaling operation to change dimensions and units.

Are you suggesting gwcs has an attribute `_supports_units` or something similar or do you see this as a `Model` attribute?

Reply:
I guess to me scaling something by a physical quantity makes sense, even if it's not a strict "scale" unless the factor is dimensionless. Do you want to send something to astropy-dev about this?

I was thinking model, but I realise now it would have to inspect if any of the parameters have units to know for certain. :fire: :wastebasket:  :fire: I guess we should take this over to gwcs to work out exactly what we need for it to work.